src/compiler/checker.ts

```diff
@@ -44588,8 +44593,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 return;
             }
 
+            const isUsedInBody = inverted && isIfStatement(condExpr.parent) ? testedSymbol && isSymbolUsedInConditionBody(location, condExpr.parent.parent, testedNode, testedSymbol, /*checkForAssignment*/ true) : testedSymbol && body && isSymbolUsedInConditionBody(condExpr, body, testedNode, testedSymbol, /*checkForAssignment*/ false);
             const isUsed = testedSymbol && isBinaryExpression(condExpr.parent) && isSymbolUsedInBinaryExpressionChain(condExpr.parent, testedSymbol)
-                || testedSymbol && body && isSymbolUsedInConditionBody(condExpr, body, testedNode, testedSymbol);
+                || isUsedInBody;
+
             if (!isUsed) {
                 if (isPromise) {

```

Comment:
Why do we call `isSymbolUsedInConditionBody` with `location` instead of `condExpr`?